src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -2108,234 +2343,6 @@ void Pet::SynchronizeLevelWithOwner()
     }
 }
 
-void Pet::HandleAsynchLoadSucceed()
-{
-    Player* owner = GetOwner();
-    if (!owner)
-        return;
-
-    if (GetAsynchLoadType() == PET_LOAD_HANDLE_UNSTABLE_CALLBACK)
-    {
-        if (Player* player = owner->ToPlayer())
-            player->GetSession()->SendStableResult(0x09 /*STABLE_SUCCESS_UNSTABLE*/);
-    }
-    else// if (GetAsynchLoadType() == PET_LOAD_BG_RESURRECT || GetAsynchLoadType() == PET_LOAD_SUMMON_PET || GetAsynchLoadType() == PET_LOAD_SUMMON_DEAD_PET)
-    {
-        // Remove Demonic Sacrifice auras (known pet)
-        Unit::AuraEffectList const& auraClassScripts = owner->GetAuraEffectsByType(SPELL_AURA_OVERRIDE_CLASS_SCRIPTS);
-        for (Unit::AuraEffectList::const_iterator itr = auraClassScripts.begin(); itr != auraClassScripts.end();)
-        {
-            if ((*itr)->GetMiscValue() == 2228)
-            {
-                owner->RemoveAurasDueToSpell((*itr)->GetId());
-                itr = auraClassScripts.begin();
-            }
-            else
-                ++itr;
-        }
-
-        SetUInt32Value(UNIT_DYNAMIC_FLAGS, UNIT_DYNFLAG_NONE);
-        RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_SKINNABLE);
-        if (GetAsynchLoadType() == PET_LOAD_SUMMON_DEAD_PET)
-        {
-            setDeathState(ALIVE);
-            ClearUnitState(uint32(UNIT_STATE_ALL_STATE & ~(UNIT_STATE_POSSESSED))); // xinef: added just in case... some linked auras and so on
-            SetHealth(CountPctFromMaxHealth(15)); // Xinef: well, only two pet reviving spells exist and both revive with 15%
-            SavePetToDB(PET_SAVE_AS_CURRENT, false);
-        }
-    }
-
-    // xinef: resurrect with full health if resurrected by BG / BF spirit
-    if (GetAsynchLoadType() == PET_LOAD_BG_RESURRECT)
-        SetHealth(GetMaxHealth());
-
-    // xinef: We are summoned in arena / battleground, remove all positive auras
-    // xinef: and set health to full if in preparation phase
-    if (GetMap()->IsBattlegroundOrArena())
-    {
-        if (GetMap()->IsBattleArena())
-            RemoveArenaAuras();
-
-        if (Player* player = owner->ToPlayer())
-            if (Battleground* bg = player->GetBattleground())
-                if (bg->GetStatus() == STATUS_WAIT_JOIN)
-                {
-                    if (IsAlive())
-                        SetHealth(GetMaxHealth());
-
-                    if (GetMap()->IsBattleground())
-                        CastSpell(this, SPELL_PREPARATION, true);
-                }
-    }
-
-    // Fix aurastate auras, depending on health!
-    // Set aurastate manualy, prevents aura switching
-    if (HealthBelowPct(20))
```

Comment:
These changes to aura states not longer needed?

Reply:
https://github.com/azerothcore/azerothcore-wotlk/blob/08d71d38d576ac75ef23b6df3bec56668b631726/src/server/game/Entities/Unit/Unit.cpp#L479